controllers/upgrade/premisesChecker_test.go

```diff
@@ -0,0 +1,218 @@
+		toDogu := readTestDataDogu(t, redmineBytes)
+		toDogu.Name = "somethingdifferent"
+
+		mockedChecker := new(premiseMock)
+		mockedChecker.On("CheckWithResource", fromDoguResource).Return(nil)
+		mockedChecker.On("ValidateDependencies", fromDogu).Return(nil)
+		mockedChecker.On("CheckDependenciesRecursive", fromDogu, fromDoguResource.Namespace).Return(nil)
+
+		sut := NewPremisesChecker(mockedChecker, mockedChecker, mockedChecker)
+
+		// when
+		err := sut.Check(ctx, fromDoguResource, fromDogu, toDogu)
+
+		// then
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "dogus must have the same name")
+		// there is no assert.IsNoType() assertion so we test it by negative type assertion
+		_, ok := err.(*requeueablePremisesError)
+		assert.False(t, ok)
+		mockedChecker.AssertExpectations(t)
+	})
+	t.Run("should fail when dependency validator fails", func(t *testing.T) {
+		fromDoguResource := readTestDataRedmineCr(t)
+		fromDogu := readTestDataDogu(t, redmineBytes)
+		toDogu := readTestDataDogu(t, redmineBytes)
+
+		mockedChecker := new(premiseMock)
+		mockedChecker.On("CheckWithResource", fromDoguResource).Return(errors.New("CheckWithResource"))
+
+		sut := NewPremisesChecker(mockedChecker, mockedChecker, mockedChecker)
+
+		// when
+		err := sut.Check(ctx, fromDoguResource, fromDogu, toDogu)
+
+		// then
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "CheckWithResource")
+		assert.IsType(t, &requeueablePremisesError{}, err)
+		// prove that the above negative type assertion works by positive type assertion
+		_, ok := err.(*requeueablePremisesError)
+		assert.True(t, ok)
+		mockedChecker.AssertExpectations(t)
+	})
+	t.Run("should fail when dogu health check fails", func(t *testing.T) {
+		fromDoguResource := readTestDataRedmineCr(t)
+		fromDogu := readTestDataDogu(t, redmineBytes)
+		toDogu := readTestDataDogu(t, redmineBytes)
+
+		mockedChecker := new(premiseMock)
+		mockedChecker.On("CheckWithResource", fromDoguResource).Return(nil)
+		mockedChecker.On("ValidateDependencies", fromDogu).Return(errors.New("ValidateDependencies"))
+
+		sut := NewPremisesChecker(mockedChecker, mockedChecker, mockedChecker)
+
+		// when
+		err := sut.Check(ctx, fromDoguResource, fromDogu, toDogu)
+
+		// then
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "ValidateDependencies")
+		assert.IsType(t, &requeueablePremisesError{}, err)
+		mockedChecker.AssertExpectations(t)
+	})
+	t.Run("should fail when dogu dependency health check fails", func(t *testing.T) {
+		fromDoguResource := readTestDataRedmineCr(t)
+		fromDogu := readTestDataDogu(t, redmineBytes)
+		toDogu := readTestDataDogu(t, redmineBytes)
+
+		mockedChecker := new(premiseMock)
+		mockedChecker.On("CheckWithResource", fromDoguResource).Return(nil)
+		mockedChecker.On("ValidateDependencies", fromDogu).Return(nil)
+		mockedChecker.On("CheckDependenciesRecursive", fromDogu, fromDoguResource.Namespace).Return(errors.New("CheckDependenciesRecursive"))
```

Comment:
You cloud use the test error object testing.AnError instead of creating a new error.